extensions/ipynb/src/notebookImagePaste.ts

```diff
@@ -0,0 +1,145 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+class CopyPasteEditProvider implements vscode.DocumentPasteEditProvider {
+
+	async provideDocumentPasteEdits(
+		_document: vscode.TextDocument,
+		_ranges: readonly vscode.Range[],
+		dataTransfer: vscode.DataTransfer,
+		_token: vscode.CancellationToken
+	): Promise<vscode.DocumentPasteEdit | undefined> {
+
+		// get foundational notebook data
+		const activeNotebook = vscode.window.activeNotebookEditor?.notebook;
```

Comment:
It would be safer to go from the `_document` (representing the current cell) to the notebook instead of assuming we are using the active notebook

Reply:
Not sure how I can use the `vscode.TextDocument` interface to get things like the current cell. I specifically need a `vscode.NotebookCell[]` object to get the metadata and attachments, unless there's something I'm missing inside `TextDocument`

Reply:
For this you can iterate through `vscode.workspace.notebooks` to find the `NotebookCell` for the passed in `vscode.TextDocument`